frame/system/src/lib.rs

```diff
@@ -1091,6 +1106,7 @@ impl<T: Trait> Module<T> {
 				},
 			}
 		);
+		Self::update_accumulators();
```

Comment:
And here?

Reply:
Ok, it should be called:

1. After initialization so that initialisation changes are not leaked as transaction changes.
2. After all storage updates within the particular transactions finished.
3. On before calculating storage root (so that finalization changes also flushed).

Reply:
If we really need all these calls for the changes trie, we don't need the accumulator. We can just keep the events in memory and write them to storage after `initialize`, `apply_transaction` and after `on_finalize`.

Reply:
We don't _really_ need changes trie imho, those calls can be later optimized and avoided with changes to changes trie so that they ignore events changes.

Changes trie should not track events, because doing so carries no information (events are changed in every transaction)

And you still need api to append something to a large storage item 🤷‍♀️

Reply:
We can also add something like `note_changed` for changes trie api if we need further speedup.